### PR TITLE
Hotfix/performace owner query load

### PR DESCRIPTION
**Pull request by RadovanTomik**

### CODE REVIEW

File: src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java [hunk 1]
```diff
@@ -98,6 +98,13 @@
                 chain.doFilter(request, response);
                 return;
             } else {
+                // If you want to view a query and you are not the owner try it as researcher
+                if( path.startsWith(req.getContextPath() +"/owner/detail.xhtml") && req.getQueryString().matches("^queryId=[0-9]*")){
+                    HttpServletResponse resp = (HttpServletResponse) response;
+                    resp.sendRedirect(req.getContextPath() + "/researcher/detail.xhtml?"+req.getQueryString());
```
[the remark this answers is not in the document]
Comment (jungwire) [reply src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java:104]: The user will be redirected to the researcher detail page of a negotiator-request, if the owner detail page is requested and the user is only a researcher and not a owner.

[the remark this answers is not in the document]
Comment (RadovanTomik) [reply src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java:104]: Code scanning does not like this approach, can you think of any other way to do the redirect?

[the remark this answers is not in the document]
Comment (jungwire) [reply src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java:104]: Sure I will look into it :)

File: src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java [hunk 2]
```diff
@@ -98,6 +98,14 @@
                 chain.doFilter(request, response);
                 return;
             } else {
+                // If you want to view a query and you are not the owner try it as researcher
+                if( path.startsWith(req.getContextPath() +"/owner/detail.xhtml") && req.getQueryString().matches("^queryId=[0-9]*")){
+                    String query_param=req.getQueryString().split("&")[0];
+                    HttpServletResponse resp = (HttpServletResponse) response;
+                    resp.sendRedirect(req.getContextPath() + "/researcher/detail.xhtml?"+query_param);
```
[the remark this answers is not in the document]
Comment (konradlang) [reply src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java:105]: Based on the warning please check the query_param to be valid, e.g. only contains "queryId=NUMBER".
Could be done using regex: /queryID=\d+/g 
Additionally the check could validate the queryID in the range of the available queryIDs from the database.

File: src/main/java/de/samply/bbmri/negotiator/control/owner/OwnerQueriesDetailBean.java [hunk 3]
```diff
@@ -274,12 +275,12 @@ public String initialize() {
 			// Get the selected(clicked on) query from the list of queries for the owner
 			// Needs to be optimized!
 
-			for(OwnerQueryStatsDTO ownerQueryStatsDTO : getQueries()) {
-				if(ownerQueryStatsDTO.getQuery().getId() == queryId) {
-					selectedQuery = ownerQueryStatsDTO.getQuery();
-					setCommentCountAndUreadCommentCount(ownerQueryStatsDTO);
-				}
-			}
+//			for(OwnerQueryStatsDTO ownerQueryStatsDTO : getQueries()) {
```
Comment (RadovanTomik) [inline src/main/java/de/samply/bbmri/negotiator/control/owner/OwnerQueriesDetailBean.java:278]: please delete it if it's no longer necessary

File: src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jungwire) [reply src/main/java/de/samply/bbmri/negotiator/filter/ResearcherOwnerFilter.java:105]: Happy to change the ^queryId=[0-9]* to ^queryId=\\d+ and/or add an additional if for query_param.
Regarding the range check, I am not sure if the ResearcherOwnerFilter is the right place for this?

File: src/main/java/de/samply/bbmri/negotiator/control/owner/OwnerQueriesDetailBean.java (hunk 3 above)
Comment (jungwire) [reply src/main/java/de/samply/bbmri/negotiator/control/owner/OwnerQueriesDetailBean.java]: Yes thangs deleted the commented code

**Review by konradlang — APPROVED** [review]

**Pull request merged**